recipes/acado/all/conanfile.py

```diff
@@ -102,11 +102,14 @@ def package_info(self):
         else:
             self.cpp_info.libs = ["acado_toolkit", "acado_casadi"]
 
+        self.cpp_info.names["CMakeDeps"] = "ACADO"
         self.cpp_info.names["cmake_find_package"] = "ACADO"
         self.cpp_info.names["cmake_find_package_multi"] = "ACADO"
 
         self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
-        self.cpp_info.build_modules.append(os.path.join("lib", "cmake", "qpoases.cmake"))
+        self.cpp_info.build_modules["CMakeDeps"].append(os.path.join("lib", "cmake", "qpoases.cmake"))
+        self.cpp_info.build_modules["cmake_find_package"].append(os.path.join("lib", "cmake", "qpoases.cmake"))
+        self.cpp_info.build_modules["cmake_find_package_multi"].append(os.path.join("lib", "cmake", "qpoases.cmake"))
```

Comment:
Are these the only changes needed for (current) conan v2?

Reply:
yes, conan does not complain anymore, the only error was:
```
ERROR: acado/1.2.2-beta: Error in package_info() method, line 109
        self.cpp_info.build_modules.append(os.path.join("lib", "cmake", "qpoases.cmake"))
        ConanV2Exception: Conan v2 incompatible: Use 'self.cpp_info.build_modules["<generator>"].append("lib/cmake/qpoases.cmake")' instead
```

Reply:
2 questions:
- this module file is not anymore in `cmake` generator, is it intended?
- what is the min conan version allowing to set CMakeDeps in cpp_info.names and self.cpp_info.build_modules?

Reply:
Your comment made me starting to browse the acado repository.
See https://github.com/acado/acado/tree/master/external_packages
Isn't qpoases a vendored external dependency?
Does acado also have a vendored eigen3?

Reply:
@SpaceIm
- regarding `cmake` generator: yes that was one of the questions in the OP, I don't want to leave out any generators, do you know of a way to add the module to all cmake basedgenerators?

@madebr 
- yes, but its a customized version that is used as an internal source.
- yes, I do not know yet if there is a way to replace it

Reply:
in 2.0 there will be only `CMakeDeps`, so other generators won't be needed (and allowed).

Reply:
So, a recipe can either be v2.0 OR pre v2.0?

Reply:
there is a transition period while old things are still allowed, at least, until 2.0 is released

Reply:
And `cmake_find_package` or a generator with similar feature will/should be kept.